editor/gui/scene_tree_editor.cpp

```diff
@@ -1735,6 +1741,8 @@ void SceneTreeDialog::set_valid_types(const Vector<StringName> &p_valid) {
 	}
 
 	show_all_nodes->show();
+	// Update the icons
+	_notification(NOTIFICATION_THEME_CHANGED);
```

Comment:
```suggestion
	if (is_inside_tree()) {
		// Force icon update.
		_notification(NOTIFICATION_THEME_CHANGED);
	}
```

Reply:
I just remembered, the reason why you don't see `_notification()` called manually anywhere is that when it's necessary there is an `_update_theme()` method, which is called in `_notification()`, but also called manually.

Not important here, but if you want to change it, you could add `_update_type_icons()` method, which would include only the loop part of theme changed.